### ISSUE TITLE
Facebook Module/Add To features non-functioning

### ISSUE THREAD

**Issue by IvanDavidLippens** [Pligg/pligg-cms#68]
Like the title says, I activated the Facebook Module today. Plopped in my key and secret, and then I got these three errors:

[15-Aug-2013 18:37:43 America/Denver] CSRF state token does not match one provided.
[15-Aug-2013 19:03:38 America/Denver] OAuthException: An active access token must be used to query information about the current user.
[15-Aug-2013 19:03:39 America/Denver] OAuthException: An active access token must be used to query information about the current user.

Before I noticed the errors, I noticed that when I selected to link my website account with my facebook account, and selected to import my avatar, nothing happened.
So first, the Avatar did not import at all.
Then I started testing various aspects of the "Post to Facebook" feature.
This is kind of hard to explain, but please bear with me.

While writing the description of the story, I noticed in the preview that it did show "Post to Facebook" and the "Add to" feature, then after submitting and moving onto the third step I noticed a little check box that allows you to check to "Post to Facebook" simultaneously upon submission. So I checked this box and clicked submit.
I checked my Facebook Wall, and nothing happened. Then I noticed that on the posts I was making there were no buttons for "Post to Facebook" or "Add To". I'm not sure when "Add To" poofed, if it was even there before, I just noticed it wasn't there then.
I made several other posts, adding links, and pictures, and turned off the third step, nothing seemed to work.
Finally I went into my apps on facebook and removed my websites app, and then went and re-logged in for facebook and granted permissions again. (The app was activated from a few weeks ago when I had been testing the Facebook app it occurred to me)

So evidently this tells me that with future facebook module updates, I might have to inform my users that they need to uninstall and reinstall the app to get features functioning to some degree.
The login feature works, to be clear. No problems there.

So when I re-established connection to the app, I tested the "Post to Facebook" features again, and this time it did post the title of the story with a URL, but it didn't even import standard OGP stuff, so it's very hokey looking and plain.
But, to be clear, there was still NO buttons. The only way I was able to do it was by the checkbox on the third confirmation page.

And also, not that it's required, but it would be nice if whenever a user connected their facebook to the pligg website account that the facebook user URL would import into the facebook section of their profile. The buttons that appear when those are set are very nice and professional looking, and I think that would be a great addition to the module.

So to summarize:
1. Avatar doesn't seem to import
2. "Post to Facebook" is semi-functional, but bugged
3. Add To is not appearing (Though I realize that this is a separate issue)
4. Facebook does not import user ID to the facebook social tab in profile settings.

The important thing, however, is that the login feature works.
I would like a ya or nay on the post to facebook feature eventually, though, because I might want to grab Chuck's Facebook Sharer in that case.

Thanks,

Ivan
